tools/partition_graph_lp.py

```diff
@@ -149,15 +156,47 @@
         target_etypes = [target_etypes]
 
     if constructed_graph:
+        d_shuffled_nids = {} # to store shuffled nids by ntype to avoid different orders for the same ntype
         for target_e in target_etypes:
             num_edges = g.num_edges(target_e)
             g.edges[target_e].data['train_mask'] = th.full((num_edges,), False, dtype=th.bool)
             g.edges[target_e].data['val_mask'] = th.full((num_edges,), False, dtype=th.bool)
             g.edges[target_e].data['test_mask'] = th.full((num_edges,), False, dtype=th.bool)
-            g.edges[target_e].data['train_mask'][: int(num_edges * args.train_pct)] = True
-            g.edges[target_e].data['val_mask'][int(num_edges * args.train_pct): \
-                                               int(num_edges * (args.train_pct + args.val_pct))] = True
-            g.edges[target_e].data['test_mask'][int(num_edges * (args.train_pct + args.val_pct)): ] = True
+            if not args.inductive_split:
+                # Randomly split links
+                g.edges[target_e].data['train_mask'][: int(num_edges * args.train_pct)] = True
+                g.edges[target_e].data['val_mask'][int(num_edges * args.train_pct): \
+                                                int(num_edges * (args.train_pct + args.val_pct))] = True
+                g.edges[target_e].data['test_mask'][int(num_edges * (args.train_pct + args.val_pct)): ] = True
+            else:
```

Comment:
i'm trying to understand why this is useful. this split is to avoid the endpoint nodes of the training/validation/test edges are not overlapped. However, this cannot guarantee that the nodes in the message passing in training and validation have overlap. do we need to avoid the overlap of nodes in the message passing?

Reply:
> However, this cannot guarantee that the nodes in the message passing in training and validation have overlap

This is true. However the tail nodes overlap is one-way: val set can contain tail nodes from training set, but training set doesn't contain tail nodes from val set. Therefore, the message passing in training set will not leak to unseen nodes in val/test sets. This also allows semi-inductive retrieval: given a new head nodes, retrieve a seen tail nodes from training set.